summingbird-online/src/main/scala/com/twitter/summingbird/planner/OnlinePlan.scala

```diff
@@ -143,10 +156,10 @@ class OnlinePlan[P <: Platform[P], V](tail: Producer[P, V]) {
           case _ if (!noOpNode(activeBolt) && dependsOnSummerProducer(currentProducer)) => true
           /*
            * This should possibly be improved, but currently, we force a FlatMapNode just before a
-           * summer (to handle map-side aggregation). This check is here to prevent us from merging
-           * this current node all the way up to the source.
+           * summer (to handle map-side aggregation) unless the currentProducer is configured to be merged into Source.
+           * This check is here to prevent us from merging this current node all the way up to the source.
            */
-          case FlatMapNode(_) if hasSummerAsDependantProducer(currentProducer) && allTransDepsMergeableWithSource(dep) => true
+          case FlatMapNode(_) if hasSummerAsDependantProducer(currentProducer) && allTransDepsMergeableWithSource(dep) && !(mergableWithSource(currentProducer)) => true
           /*
```

Comment:
+1 to that.

I still think you have to split if the node has some summer and some non-summer dependant. Willingness to merge with a source is not enough (mergeableWithSource) it has to also be correct to do so.

```scala
val n1 = src.flatMap(fn)
val n2 = n1.sumByKey(store)
val n3 = n1.leftJoin(srv).write(sink)
n3.also(n2)
```
In that case `n1` cannot be in the source because though it has a Summer dependant and it might be mergeableWithSource, the `leftJoin` might be messed up if we did that.